### PR TITLE
Cleanup and optimize Helm charts

### PR DESCRIPTION
**Pull request by heoelri**
This PR superseeds PR #356 and #351. See #356 for more details. 

https://dev.azure.com/Mission-Critical/Online/_build/results?buildId=1552&view=results

### CODE REVIEW

File: src/app/charts/backgroundprocessor/templates/deployment.yaml [hunk 1]
```diff
@@ -35,13 +34,9 @@ spec:
...
           initialDelaySeconds: 5
           periodSeconds: 10
-        resources:
-          limits:
-            cpu: "0.5"
-            memory: "1Gi"
-          requests:
-            cpu: "0.5"
-            memory: "512Mi"
+        {{- if .Values.resources }}
```
Comment (sebader) [inline src/app/charts/backgroundprocessor/templates/deployment.yaml:37]: should this even have an if? When this is not specified, deployment should fail

Comment (heoelri) [reply src/app/charts/backgroundprocessor/templates/deployment.yaml:37]: The currently available policies do not really solve that. You can specify limits, but you cannot make `resources` mandatory. For now I'd like to keep it as it is. Our helm charts are considered to follow "best practices" and I'd not rely on any existing or not existing expressions inside the helm chart to fail a deployment.

Especially as no one would prevent a user from removing resources from the `values.yaml` as well as from the `templates/deployment.yaml`.

**Comment by heoelri** [comment]
Needs to be ported to connected as well.

File: .ado/pipelines/templates/jobs-workload-deploy.yaml [hunk 2]
```diff
@@ -74,18 +107,15 @@ jobs: # using jobs so they each run in parallel
           # load json data from downloaded terraform artifact
           $releaseUnitInfraDeployOutput = Get-ChildItem $(Pipeline.Workspace)/terraformOutputReleaseUnitInfra/*.json | Get-Content | ConvertFrom-JSON
 
-          echo "*** Retrieved Azure Front Door Header ID $(frontdoor_id_header) from Terraform output"
```
Comment (heoelri) [inline .ado/pipelines/templates/jobs-workload-deploy.yaml:77]: @sebader was it intended to drop this? I think this is valuable debug output.

Comment (sebader) [reply .ado/pipelines/templates/jobs-workload-deploy.yaml:77]: yeah I realized we never really needed this one (or never had issues). you can always just look it up in the artifact. and it is kind of a secret so not logging it is probably better

**Pull request merged**